lib/sqlmap/sqlmap_manager.rb

```diff
@@ -8,43 +8,56 @@ def initialize(session)
 
     def new_task
       res = @session.get('/task/new')
-      return JSON.parse(res.body)
+      parse_response(res)
     end
 
     def delete_task(task_id)
       res = @session.get('/task/' + task_id + '/delete')
-      return JSON.parse(res.body)
+      parse_response(res)
     end
 
     def set_option(task_id, key, value)
       post = { key => value }
       res = @session.post('/option/' + task_id + '/set', nil, post.to_json, {'ctype' => 'application/json'})
-      return JSON.parse(res.body)
+      parse_response(res)
     end
 
     def get_options(task_id)
       res = @session.get('/option/' + task_id + '/list')
-      return JSON.parse(res.body)
+      parse_response(res)
     end
 
     def start_task(task_id, options = {})
       res = @session.post('/scan/' + task_id + '/start' , nil, options.to_json, {'ctype' => 'application/json'})
-      return JSON.parse(res.body)
+      parse_response(res)
+
     end
 
     def get_task_status(task_id)
       res = @session.get('/scan/' + task_id + '/status')
-      return JSON.parse(res.body)
+      parse_response(res)
     end
 
     def get_task_log(task_id)
       res = @session.get('/scan/' + task_id + '/log')
-      return JSON.parse(res.body)
+      parse_response(res)
     end
 
     def get_task_data(task_id)
       res = @session.get('/scan/' + task_id + '/data')
-      return JSON.parse(res.body)
+      parse_response(res)
+    end
+
+    private
+    def parse_response(res)
+      if res && res.body
+        begin
+          res = JSON.parse(res.body)
+        rescue JSON::ParserError
+        end
+      end
+
+      res
```

Comment:
In my opinion ideally one method should only return one return value type, otherwise it's confusing to use (like when people use send_request_cgi, they tend forget it doesn't always return a response object, it could return a NilClass too).

I suggest doing the following:

```ruby
# Returns the JSON response.
#
# @param [Rex::Proto::Http::Response] res An HTTP response object.
# @return [Hash] JSON data.
def parse_response(res)
  json = {}

  begin
    json = JSON.parse(res.body)
  rescue JSON::ParserError
  end

  json
end
```

And then whatever API you're doing, do something like:

```ruby
# taskid is always a hash, but you can check taskid.empty? before accessing a key if you want.
taskid = @manager.new_task
if taskid['taskid']
  # Do something with taskid
else
  print_error("No JSON data.")
end
```